packages/contracts-bedrock/test/L2/L2ToL2CrossDomainMessenger.t.sol

```diff
@@ -744,4 +744,34 @@ contract L2ToL2CrossDomainMessengerTest is Test {
         // Call `crossDomainMessageSource` to provoke revert
         l2ToL2CrossDomainMessenger.crossDomainMessageSource();
     }
+
+    /// @dev Tests that the `crossDomainMessageContext` function returns the correct value.
+    function testFuzz_crossDomainMessageContext_succeeds(address _sender, uint256 _source) external {
+        // Set `entered` to non-zero value to prevent NotEntered revert
+        l2ToL2CrossDomainMessenger.setEntered(1);
+        // Ensure that the contract is now entered
+        assertEq(l2ToL2CrossDomainMessenger.entered(), true);
+
+        // Set cross domain message source in the transient storage
+        l2ToL2CrossDomainMessenger.setCrossDomainMessageSender(_sender);
+        l2ToL2CrossDomainMessenger.setCrossDomainMessageSource(_source);
+
+        // Check that the `crossDomainMessageContext` function returns the correct value
+        (address crossDomainContextSender, uint256 crossDomainContextSource) =
+            l2ToL2CrossDomainMessenger.crossDomainMessageContext();
+        assertEq(crossDomainContextSender, _sender);
+        assertEq(crossDomainContextSource, _source);
+    }
+
+    /// @dev Tests that the `crossDomainMessageContext` function reverts when not entered.
+    function test_crossDomainMessageContext_notEntered_reverts() external {
```

Comment:
Curious why this is the behavior rather than returning zero values?

Reply:
We replicated the logic from `crossDomainMessageSender()` and `crossDomainMessageSource()`. We also noticed that the same modifier is used in `CrossL2Inbox`, so it might make sense to maintain this as a standard.

We believe it's a good idea to have them revert if called outside of a message context, as this makes it extra clear that these functions are only useful within that specific context.

Reply:
> We believe it's a good idea to have them revert if called outside of a message context, as this makes it extra clear that these functions are only useful within that specific context.

I think that's the right approach, as it avoids a silent failure.